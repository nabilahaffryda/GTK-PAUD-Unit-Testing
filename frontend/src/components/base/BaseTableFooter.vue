<template>
  <div class="text-center" style="width: 100%">
    <template v-if="useNumber">
      <v-pagination
        :circle="circle"
        color="secondary"
        v-model="page"
        :length="pageTotal"
        prev-icon="mdi-chevron-left"
        next-icon="mdi-chevron-right"
        :total-visible="5"
      >
      </v-pagination>
    </template>
    <template v-else>
      <v-btn
        fab
        color="secondary"
        class="ma-1"
        x-small
        :disabled="Number(page) === 1"
        :depressed="Number(page) === 1"
        @click="prev"
      >
        <v-icon>mdi-chevron-left</v-icon>
      </v-btn>
      <v-btn
        fab
        color="secondary"
        class="ma-1"
        x-small
        :disabled="Number(total) < 10"
        :depressed="Number(total) < 10"
        @click="next"
      >
        <v-icon>mdi-chevron-right</v-icon>
      </v-btn>
    </template>
  </div>
</template>

<script>
export default {
  name: 'BaseTableFooter',
  props: {
    pageTotal: {
      type: Number,
      default: 0,
    },
    total: {
      type: Number,
      default: 0,
    },
    circle: {
      type: Boolean,
      default: true,
    },
    useNumber: {
      type: Boolean,
      default: true,
    },
  },
  data() {
    return {
      page: 1,
    };
  },
  methods: {
    next() {
      this.page = this.page + 1;
    },
    prev() {
      this.page = this.page > 1 ? this.page - 1 : this.page;
    },
    changePage() {
      this.$emit('changePage', this.page);
    },
  },
  watch: {
    page: 'changePage',
  },
};
</script>

<style scoped></style>
