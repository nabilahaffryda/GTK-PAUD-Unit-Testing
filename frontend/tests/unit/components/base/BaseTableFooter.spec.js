import { mount, createLocalVue } from "@vue/test-utils";
import BaseTableFooter from "@/components/base/BaseTableFooter.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';
import VueRouter from 'vue-router';

Vue.use(Vuetify)
const localVue = createLocalVue();
localVue.use(VueRouter)
const router = new VueRouter();

describe('BaseTableFooter.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    function wrapperFactory({ } = {}) {
        return mount(BaseTableFooter, {
            localVue,
            vuetify,
            router,
        });
    }
    test('call next method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.next = jest.fn();
        wrapper.vm.next();
        expect(wrapper.vm.next.mock.calls.length).toBe(1);
    })
    test('call prev method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.prev = jest.fn();
        wrapper.vm.prev();
        expect(wrapper.vm.prev.mock.calls.length).toBe(1);
    })
    test('call changePage method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.changePage = jest.fn();
        wrapper.vm.changePage();
        expect(wrapper.vm.changePage.mock.calls.length).toBe(1);
    })
})
