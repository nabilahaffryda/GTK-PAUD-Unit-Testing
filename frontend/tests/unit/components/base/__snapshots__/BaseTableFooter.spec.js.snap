// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTableFooter.vue render table footer 1`] = `
<div class="text-center" style="width: 100%;">
  <nav role="navigation" aria-label="Pagination Navigation">
    <ul class="v-pagination v-pagination--circle theme--light">
      <li><button disabled="disabled" type="button" aria-label="Previous page" class="v-pagination__navigation v-pagination__navigation--disabled"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></button></li>
      <li><button disabled="disabled" type="button" aria-label="Next page" class="v-pagination__navigation v-pagination__navigation--disabled"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></button></li>
    </ul>
  </nav>
</div>
`;
